Source/sources/Castro_sources.cpp

```diff
@@ -466,29 +466,36 @@ Castro::print_all_source_changes(Real dt, bool is_new)
 // Obtain the sum of all source terms.
 
 void
-Castro::sum_of_sources(MultiFab& source)
+Castro::make_sdc_hydro_plus_sources(MultiFab& source, Real dt)
 {
-  BL_PROFILE("Castro::sum_of_sources()");
+  BL_PROFILE("Castro::make_sdc_hydro_plus_sources()");
 
   // this computes advective_source + 1/2 (old source + new source)
   //
-  // Note: the advective source is defined as -div{F}
   //
-  // the time-centering is accomplished since new source is defined
-  // to be 1/2 (new source - old source) generally.
+  // we have the following:
+  //
+  // S_new :      U^n - dt div{F} + dt/2 (S^n + S^{n+1})
+  // Sborder:     U^n
```

Comment:
Can you use S_old here instead of Sborder? That way if we ever do get rid of Sborder this won't break. S_old should be equivalent to Sborder for simplified SDC, and has the same number of ghost zones as S_new, so this should generally work.

Reply:
I suppose I could, will they both have had a clean_state?

Reply:
Yes, we have to do clean_state on S_old at the beginning of the advance because we may have data filled in via AMR that won't be thermodynamically consistent.

Reply:
okay, updated